### ISSUE TITLE
ONNX export regression in main (NVIDIA pytorch container 22.03)

### ISSUE THREAD

**Issue by borisfom** [pytorch/pytorch#75004]
### 🐛 Describe the bug

A number of nets that exported fine with NVIDIA pytorch container 22.02, are now failing to export with diagnostic: 
```
               params_dict = torch._C._jit_pass_onnx_deduplicate_initializers(graph, params_dict,
                                                                               training == TrainingMode.TRAINING)
E                                                                              RuntimeError: Expected all tensors to be on the same device, but found at least two devices, cpu and cuda:0! (when checking argument for argument other in method wrapper__equal)

```
I was able to track one of those error down and fixed it with the diff below. 
It appears exporter gets confused about device placement of inline constants (?) as well as class attributes that are scalars, not proper Variables. Below, both literal 0 and self.max_token_duration had to be moved to buffers to be exported properly:

```
-        self.max_token_duration = max_token_duration
+        self.register_buffer('max_token_duration', torch.tensor(max_token_duration))
+        self.register_buffer('min_token_duration', torch.tensor(0.0))
...
-        durs_predicted = torch.clamp(torch.exp(log_durs_predicted) - 1, 0, self.max_token_duration)
+        durs_predicted = torch.clamp(
+            torch.exp(log_durs_predicted) - 1.0, self.min_token_duration, self.max_token_duration
+        )

```
I think this is a bug, as most operations do not have those issues with literals and non-buffer attributes.

### Versions

Trunk; was fine a month ago (NVIDIA pytorch container 22.02)

**Comment by borisfom** [comment]
Also: this issue causes latest transformers to fail : https://github.com/huggingface/transformers/issues/16532

**Comment by BowenBao** [comment]
Thanks for reporting this regression. Created a pr to fix it #75273

**Comment by MMMMMax** [comment]
for key in params_dict:
                params_dict[key] = params_dict[key].to('cpu')
![image](https://github.com/pytorch/pytorch/assets/37059690/c8874042-55a4-4720-b4b0-c805a6bbe26a)
